skfda/preprocessing/dim_reduction/feature_extraction/_per_class_transformer.py

```diff
@@ -0,0 +1,280 @@
+
+        Check that it is scikit-learn-like and that
+        uses target data in fit.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        if not (
+            hasattr(self.transformer, "fit")  # noqa: WPS421
+            and hasattr(self.transformer, "transform")  # noqa: WPS421
+            and hasattr(self.transformer, "fit_transform")  # noqa: WPS421
+        ):
+            raise TypeError(
+                "Transformer should implement fit and "
+                "transform. " + str(self.transformer)
+                + " (type " + str(type(self.transformer)) + ")"
+                " doesn't",
+            )
+
+        tags = self.transformer._get_tags()  # noqa: WPS437
+
+        if tags['stateless'] or not tags['requires_y']:
+            warnings.warn(
+                f"Parameter ``transformer`` with type"  # noqa: WPS237
+                f" {type(self.transformer)} should use class information."
+                f" It should have the ``requires_y`` tag set to ``True`` and"
+                f" the ``stateless`` tag set to ``False``",
+            )
+
+    def fit(
+        self,
+        X: Input,
+        y: np.ndarray,
+    ) -> PerClassTransformer[Input, Output, Target]:
+        """
+        Fit the model on each class.
+
+        It uses X as training data and y as target values.
+
+        Args:
+            X: FDataGrid with the training data.
+            y: Target values of shape = (n_samples).
+
+        Returns:
+            self
+        """
+        self._validate_transformer()
+        classes, class_feature_transformers = _fit_feature_transformer(
+            X,
+            y,
+            self.transformer,
+        )
+
+        self._classes = classes
+        self._class_feature_transformers_ = class_feature_transformers
+
+        return self
+
+    def transform(self, X: Input) -> Union[DataFrame, np.ndarray]:
+        """
+        Transform the provided data using the already fitted transformer.
+
+        Args:
+            X: FDataGrid with the test samples.
+
+        Returns:
+            Eiter array of shape (n_samples, G) or a Data Frame
+            including the transformed data.
+        """
+        sklearn_check_is_fitted(self)
+
+        transformed_data = [
+            feature_transformer.transform(X)
+            for feature_transformer in self._class_feature_transformers_
+        ]
+
+        if self.array_output:
+            for i, data in enumerate(transformed_data):
+                if isinstance(data, (FDataGrid, FDataBasis)):
+                    raise TypeError(
+                        "There are transformed instances of FDataGrid or "
+                        "FDataBasis that can't be concatenated on a NumPy "
+                        "array.",
+                    )
+                elif i == 0:
+                    transformed_array = data
+                else:
+                    transformed_array = np.hstack((transformed_array, data))
```

Comment:
Can't you do `np.array(transformed_data).T`? Concatenating arrays is absurd if you already have all the data.

Reply:
The error still appears.
When executing pytest, the first example shows: "ValueError('could not broadcast input array from shape (93,24) into shape (93,)')" .
It seems that the arrays contained in transformed_data have sizes (93, 24) and (93, 18) and that is why it gives an exception when creating a ndarray.